src/commands/Ban.tsx

```diff
@@ -0,0 +1,161 @@
+ * are NOT distributed, contributed, committed, or licensed under the Apache License.
+ */
+
+import { MatrixRoomReference } from "./interface-manager/MatrixRoomReference";
+import { UserID } from "matrix-bot-sdk";
+import { MjolnirContext } from "./CommandHandler";
+import { CommandError, CommandResult } from "./interface-manager/Validation";
+import { Mjolnir } from "../Mjolnir";
+import { RULE_ROOM, RULE_SERVER, RULE_USER } from "../models/ListRule";
+import PolicyList from "../models/PolicyList";
+import { defineInterfaceCommand, findTableCommand } from "./interface-manager/InterfaceCommand";
+import { findPresentationType, makePresentationType, ParamaterDescription, paramaters, ParsedKeywords, RestDescription, simpleTypeValidator, union } from "./interface-manager/ParamaterParsing";
+import "./interface-manager/MatrixPresentations";
+import { defineMatrixInterfaceAdaptor } from "./interface-manager/MatrixInterfaceAdaptor";
+import { tickCrossRenderer } from "./interface-manager/MatrixHelpRenderer";
+import { PromptOptions } from "./interface-manager/PromptForAccept";
+import { definePresentationRenderer } from "./interface-manager/DeadDocumentPresentation";
+import { DocumentNode } from "./interface-manager/DeadDocument";
+import { JSXFactory } from "./interface-manager/JSXFactory";
+
+makePresentationType({
+    name: "PolicyList",
+    validator: simpleTypeValidator("PolicyList", (readItem: unknown) => readItem instanceof PolicyList)
+})
+
+definePresentationRenderer(findPresentationType("PolicyList"), function(list: PolicyList): DocumentNode {
+    return <p>
+        {list.listShortcode} <a href={list.roomRef}>{list.roomId}</a>
+    </p>
+})
+
+
+export async function findPolicyListFromRoomReference(mjolnir: Mjolnir, policyListReference: MatrixRoomReference): Promise<CommandResult<PolicyList, CommandError>> {
+    // do we need to catch 404 here so we can tell the user the alias was wrong?
+    // Can we shortcut this by making a specific presentaion type for
+    // resolved room id? (that aliases can translate to)
+    const policyListRoomId = (await policyListReference.resolve(this.client)).toRoomIdOrAlias();
+    const policyList = mjolnir.policyListManager.lists.find(list => list.roomId === policyListRoomId);
+    if (policyList !== undefined) {
+        return CommandResult.Ok(policyList);
+    } else {
+        // Would it be acceptable to create an anonymous policy list that is not being watched
+        // by mjolnir for the purposes of banning / unbanning? unbanning requires loading the rules
+        // but banning doesn't.. so this means you'd want two types of lists
+        // one of which can only be made by a factory which watches them via sync
+        // and makes sure mjolnir is joined.
+        // This refactor would be important for previewing lists regardless.
+        return CommandError.Result(`There is no policy list that Mjolnir is watching for ${policyListReference.toPermalink()}`);
```

Comment:
if the policy list isn't being watched, create an anonymous list. When unbanning do the same but call `list.update` before calling `unban`.